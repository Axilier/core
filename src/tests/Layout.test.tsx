/** @format */

import { mount } from 'enzyme';
import React from 'react';
import { Layout } from '../index';

const layoutColumn = mount(<Layout orientation={'column'}>click me</Layout>);
const layoutRow = mount(<Layout orientation={'row'}>click me</Layout>);

describe('when the `orientation` prop is set to row', () => {
    it('should render correctly', () => {
        expect(layoutRow).toMatchSnapshot();
    });
    it('should have the expected styles', () => {
        expect(layoutRow.childAt(0).props().style).toHaveProperty(
            'flexDirection',
            'row',
        );
    });
});
describe('when the `orientation` prop is set to column', () => {
    it('should render correctly', () => {
        expect(layoutColumn).toMatchSnapshot();
    });

    it('should have the expected styles', () => {
        expect(layoutColumn.childAt(0).props().style).toHaveProperty(
            'flexDirection',
            'column',
        );
    });
});
describe('when the `orientation` prop is unset', () => {
    beforeAll(() => layoutColumn.setProps({ orientation: undefined }));
    it('should have the expected styles', () => {
        expect(layoutColumn.childAt(0).props().style).toHaveProperty(
            'flexDirection',
            'row',
        );
    });
    afterAll(() => layoutColumn.setProps({ orientation: 'column' }));
});
describe('when the `center` prop is true', () => {
    beforeAll(() => layoutColumn.setProps({ center: true }));
    it('it should have styles to center its child content', () => {
        expect(layoutColumn.childAt(0).props().style).toHaveProperty(
            'alignItems',
            'center',
        );
        expect(layoutColumn.childAt(0).props().style).toHaveProperty(
            'justifyContent',
            'center',
        );
    });
    afterAll(() => layoutColumn.setProps({ center: false }));
});
describe('when the `center` prop is not set', () => {
    it('it should have styles unset center styles', () => {
        expect(layoutColumn.childAt(0).props().style).not.toHaveProperty(
            'alignItems',
        );
        expect(layoutColumn.childAt(0).props().style).not.toHaveProperty(
            'justifyContent',
        );
    });
});
