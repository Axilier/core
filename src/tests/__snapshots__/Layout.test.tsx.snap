// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`when the \`orientation\` prop is set to column should render correctly 1`] = `
<Layout
  orientation="column"
>
  <div
    style={
      Object {
        "display": "flex",
        "flexDirection": "column",
      }
    }
  >
    click me
  </div>
</Layout>
`;

exports[`when the \`orientation\` prop is set to row should render correctly 1`] = `
<Layout
  orientation="row"
>
  <div
    style={
      Object {
        "display": "flex",
        "flexDirection": "row",
      }
    }
  >
    click me
  </div>
</Layout>
`;
